server/src/main/java/org/elasticsearch/action/search/InitialSearchPhase.java

```diff
@@ -232,10 +224,18 @@ private synchronized Runnable tryQueue(Runnable runnable) {
     }
 
     private void executeNext(PendingExecutions pendingExecutions, Thread originalThread) {
+        executeNext(pendingExecutions == null ? null : pendingExecutions::finishAndRunNext, originalThread);
+    }
+
+    protected void executeNext(Runnable runnable, Thread originalThread) {
```

Comment:
I added this protected method rather than making the existing `executeNext` protected, just to leave the `PendingExecutions` class private. While I was at it, I folded `maybeFork` into this one as it was only called from here anyways.